google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/StreamingSubscriberConnection.java

```diff
@@ -211,35 +211,45 @@ private boolean isAlive() {
   @Override
   public void sendAckOperations(
       List<String> acksToSend, List<PendingModifyAckDeadline> ackDeadlineExtensions) {
+    StreamingPullRequest.Builder[] requests =
+        partitionAckOperations(acksToSend, ackDeadlineExtensions, MAX_PER_REQUEST_CHANGES);
+    for (StreamingPullRequest.Builder request : requests) {
+      requestObserver.onNext(request.build());
```

Comment:
It seems odd to have `partitionAckOperations` return an array of builders when the only thing it does is build them here. Why not call `build()` in `partitionAckOperations`?